docs/biscuit-auth.md

```diff
@@ -0,0 +1,245 @@
+
+For example, to grant read access to peers, write access to payments, and add an expiration date:
+
+```datalog
+// Grant permissions for specific modules
+read("peers");
+write("payments");
+
+// You can also add checks, like an expiration date.
+// This check ensures the token is only valid before the specified UTC timestamp.
+check if time($time), $time <= 2025-01-01T00:00:00Z;
+```
+
+### b. Generate the Token
+
+Use the `biscuit generate` command with your private key and the permissions file to create the token.
+
+```bash
+biscuit generate --private-key ed25519-private/89d6c88919e5ca326fbb8d1cbef406df08c0620575376651d53008762dc81f45 permissions.bc
+```
+
+The command will output a long Base64-encoded string. This is your bearer token.
+
+```
+ErsBClEKBXBlZXJzCghwYXltZW50cxgDIgkKBwgAEgMYgAgiCQoHCAESAxiBCDImCiQKAggbEgYIBRICCAUaFgoECgIIBQoICgYggIvSuwYKBBoCCAISJAgAEiDAjoKKNTZpA61ImgoD5Q2sSbBjA3ixK1R2M65a2TOxbRpA4SF04LS5zD6OempqQObA2TTlTCANI7bZkDpl7JIecjR59GFoNtyKYak-_jXq2gDUHadj2I8SbJ0N-vN1RPslDSIiCiBR7VZ7ZJaPE4nhxUUpgpd5MXr3hqi0RfNERy3NE4KjaQ==
+```
+
+### c. Using the Token
+
+You can now use this token in the `Authorization` header when making RPC calls to your Fiber node:
+
+```
+Authorization: Bearer ErsBClEKBXBlZXJzCghwYXltZW50cxgDIgkKBwgAEgMYgAgiCQoHCAESAxiBCDImCiQKAggbEgYIBRICCAUaFgoECgIIBQoICgYggIvSuwYKBBoCCAISJAgAEiDAjoKKNTZpA61ImgoD5Q2sSbBjA3ixK1R2M65a2TOxbRpA4SF04LS5zD6OempqQObA2TTlTCANI7bZkDpl7JIecjR59GFoNtyKYak-_jXq2gDUHadj2I8SbJ0N-vN1RPslDSIiCiBR7VZ7ZJaPE4nhxUUpgpd5MXr3hqi0RfNERy3NE4KjaQ==
+```
+
+This token grants exactly the permissions you defined and will be successfully verified by a Fiber server configured with the corresponding public key.
+
+## 5. Test Examples
+
+The Fiber codebase contains numerous tests that demonstrate the usage of Biscuit tokens. These are excellent resources for understanding the system.
+
+### a. RPC Integration Tests
+
+The file `crates/fiber-lib/src/fiber/tests/rpc.rs` contains integration-style tests for the RPC interface. Key examples include:
+
+*   `test_rpc_basic_with_auth`: Shows how to make multiple authenticated RPC calls with a single token.
+*   `test_rpc_auth_without_token`: Demonstrates that an unauthenticated request to a protected endpoint is rejected.
+*   `test_rpc_auth_with_invalid_token`: Tests that a token signed by an incorrect private key is rejected.
+*   `test_rpc_auth_with_wrong_permission`: Shows that a valid token with insufficient permissions is correctly denied access.
+
+[Link to file: `crates/fiber-lib/src/fiber/tests/rpc.rs`](../crates/fiber-lib/src/fiber/tests/rpc.rs)
+
+### b. Unit Tests for Authorization Logic
+
+The file `crates/fiber-lib/src/rpc/biscuit.rs` contains unit tests that focus specifically on the token authorization logic. These are useful for seeing how different kinds of Datalog rules are handled.
```

Comment:
this is not useful to refer in documentation, since they are only internal tests, not about use model.

Reply:
The tests can suggest how to sign biscuit token, and how to write biscuit rules.